2D3D_UnityProject/Assets/Puzzles/Liars/LiarGameManager.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class LiarGameManager : MonoBehaviour
+{
+    /// <summary>
+    /// reference to the canvas
+    /// </summary>
+    public GameObject canvasObject;
+
+    /// <summary>
+    /// reference to the DialogueLine script
+    /// </summary>
+    private DialogueLine dialogueCode;
+
+    /// <summary>
+    /// list of all npcs you can talk to in the scene
+    /// </summary>
+    public GameObject[] npcs;
+
+    /// <summary>
+    /// reference to player
+    /// </summary>
+    public GameObject player;
+
+    /// <summary>
+    /// instance of this manager
+    /// </summary>
+    private static LiarGameManager instance;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        if (instance == null)
+            instance = this;
```

Comment:
Yes singleton fuck yeah!!!

I actually brought in a Singleton class that does this boilerplate singleton setup for you - I can show you how to use it later. Very glad to see this used here - makes sense for a puzzle manager class